pkg/repstream/streampb/stream.proto

```diff
@@ -66,15 +66,25 @@ message StreamPartitionSpec {
     google.protobuf.Duration min_checkpoint_frequency = 2
        [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
 
-    // Controls batch size in bytes.
+    // Controls the rangefeed batch size in bytes.
     int64 batch_byte_size = 3;
-
-    roachpb.TenantID span_config_for_tenant = 4 [(gogoproto.nullable) = false];
   }
 
   ExecutionConfig config = 3 [(gogoproto.nullable) = false];
 }
 
+// SpanConfigEventStreamSpec is the span config event stream specification.
+message SpanConfigEventStreamSpec {
+  roachpb.Span span = 1 [(gogoproto.nullable) = false];
+
```

Comment:
Do we need this?

Reply:
seems like, we could move all the logic in this `setupSpanConfigStream` function into start, and just pass the `tenantName` if we really wanted to.

Reply:
I think the thing that rubs me the wrong way about it is having the consumer passing around a span that isn't contained inside the tenant's keyspace.  But, what you have here is a strict improvement over the old code so we don't need to block on it.

Reply:
I see. I may need to refactor this specific part of the code to address this todo anyway, so i'd like to just merge what I have now:

https://github.com/cockroachdb/cockroach/blob/fcf26502c76ff0b7f8e04e31f4fc9b466d3cf53d/pkg/ccl/streamingccl/streamproducer/stream_lifetime.go#L359

Also, if it makes you feel any better, the only way for the consumer to access these fields would be to cast the `ValueGenerator` into a `spanConfigEventStream` which I think will always store state about producer side system tenant key space.